tools/benchmark_checking.py

```diff
@@ -1,70 +1,64 @@
-    sys.exit(1)
-
-try:
-    os.remove('.ses_state')
-except Exception:
-    pass
-try:
-    shutil.rmtree('.resume')
-except Exception:
-    pass
-
-if not os.path.exists('checking_benchmark.torrent'):
-    ret = os.system('../examples/connection_tester gen-torrent -s 10000 -n 15 -t checking_benchmark.torrent')
-    if ret != 0:
-        print('ERROR: connection_tester failed: %d' % ret)
-        sys.exit(1)
 
-if not os.path.exists('checking_benchmark'):
-    ret = os.system('../examples/connection_tester gen-data -t checking_benchmark.torrent -p .')
+def main():
+    args = parse_args()
+
+    ret = os.system('cd ../examples && b2 release %s stage_client_test stage_connection_tester'
+                    % args.toolset)
     if ret != 0:
-        print('ERROR: connection_tester failed: %d' % ret)
+        print('ERROR: build failed: %d' % ret)
         sys.exit(1)
 
+    rm_file_or_dir(".ses_state")
+    rm_file_or_dir(".resume")
+
+    if not os.path.exists('checking_benchmark.torrent'):
+        ret = os.system('../examples/connection_tester gen-torrent -s 10000 -n 15 -t checking_benchmark.torrent')
+        if ret != 0:
+            print('ERROR: connection_tester failed: %d' % ret)
+            sys.exit(1)
+
+    if not os.path.exists('checking_benchmark'):
+        ret = os.system('../examples/connection_tester gen-data -t checking_benchmark.torrent -p .')
+        if ret != 0:
+            print('ERROR: connection_tester failed: %d' % ret)
+            sys.exit(1)
+
+    for threads in [4, 8, 16, 32, 64]:
+        run_test('%d' % threads, '--hashing_threads=%d' % threads)
+
 
 def run_test(name, client_arg):
     output_dir = 'logs_checking_%s' % name
 
-    if os.path.exists(output_dir) and os.path.exists(os.path.join(output_dir, 'timing.txt')):
+    timing_path = os.path.join(output_dir, 'timing.txt')
+    if os.path.exists(timing_path):
+        print('file "{path}" exists, skipping test "{name}"'.format(path=timing_path, name=name))
         return
 
-    try:
-        shutil.rmtree(output_dir)
-    except Exception:
-        pass
+    rm_file_or_dir(output_dir)
     try:
         os.mkdir(output_dir)
     except Exception:
         pass
 
-    try:
-        shutil.rmtree(".resume")
-    except Exception:
-        pass
+    rm_file_or_dir(".resume")
 
-    client_cmd = '../examples/client_test checking_benchmark.torrent ' \
-                 '--enable_dht=0 --enable_lsd=0 --enable_upnp=0 --enable_natpmp=0 ' \
-                 '-1 %s -s . -f %s/events.log --alert_mask=all' \
-                 % (client_arg, output_dir)
+    client_cmd = ('../examples/client_test checking_benchmark.torrent '
+                  '--enable_dht=0 --enable_lsd=0 --enable_upnp=0 --enable_natpmp=0 '
+                  '-1 %s -s . -f %s/events.log --alert_mask=all'
+                  ) % (client_arg, output_dir)
 
     client_out = open('%s/client.out' % output_dir, 'w+')
-    print(client_cmd)
+    print('client_cmd: "{cmd}"'.format(cmd=client_cmd))
```

Comment:
I think this could even be simplified to:
```
print(f'client_cmd: "{client_cmd}"')
```

Reply:
You can also further simply it to:

```python3
print(f"{client_cmd=}")
```

With python >= 3.8. Optionally, you can have spaces around the `=`, which will also appear in the output.

Reference: https://docs.python.org/3/whatsnew/3.8.html#f-strings-support-for-self-documenting-expressions-and-debugging

Reply:
Yeah, good point @arvidn :) The primary reason I did not use literal string interpolation is that I didn't want to impose a python version (python 3.5 future or 3.6).

I did not know about the syntax you just posted at all, @FranciscoPombal, so thanks for that :)

Reply:
I'll merge it and fix it up to pass CI in master then